processor/factory.go

```diff
@@ -50,6 +55,24 @@ type Factory interface {
 		cfg configmodels.Processor) (MetricsProcessor, error)
 }
 
+// FactoryV2 is factory interface for processors. This is the
+// new factory type that can create new style processors.
+type FactoryV2 interface {
+	BaseFactory
+
+	// CreateTraceProcessorV2 creates a trace processor based on this config.
+	// If the processor type does not support tracing or if the config is not valid
+	// error will be returned instead.
+	CreateTraceProcessorV2(logger *zap.Logger, nextConsumer consumer.TraceConsumerV2,
```

Comment:
This, and the related interfaces/types, should accept a context as the first parameter, just like their receiver counterparts (`CreateTraceReceiverV2`, `CreateMetricReceiverV2`)

Reply:
Can you clarify? What do you expect the context to be used for in creation functions?

Reply:
Hard to say upfront, but one possible case is a processor that reads a configuration from an external resource at initialization. A cancellation/deadline from the parent should then be propagated to the processors.

Reply:
`Create*` functions are not expected to perform actual initialization of the components. `Start` is intended for that. Any long-lasting or potentially blocking operation should happen in `Start` and context is already available in `Start` via `Host.Context()`.

Looking at the current source code the context is not used to any of current `Create*` functions that accept it (everyone just uses context.Background() at the callsite and non of `Create*` implementations does anything with the passed context).

However we likely need to refine our startup cancellation strategy. I am fine with adding the context since it give us the possibility to refine our strategy in the future without introducing breaking changes in exported interfaces. This way we will be more future-proof.

@open-telemetry/collector-maintainers what do you think?

Reply:
We should recommend that components do not get any resource other than memory at creation time: it makes clean shut down for this case harder. That said, passing the context seems reasonable, per the reasons above (also opening door to some observability using context).

Reply:
> also opening door to some observability using context

+1. Personally, I always use `context` when I expect the method to participate in the deadline or when I'd possibly benefit from wrapping the call in a span. Might not apply to OpenTelemetry itself, though ;-)

Reply:
@tigrannajaryan log correlation may be another example :), so I think having context is not bad.

Reply:
Done.